src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -228,18 +228,22 @@ def _encode_authority_information_access(backend, authority_info_access):
     return pp, r
 
 
-def _encode_subject_alt_name(backend, san):
+def _encode_general_names(backend, names):
     general_names = backend._lib.GENERAL_NAMES_new()
     assert general_names != backend._ffi.NULL
-    general_names = backend._ffi.gc(
-        general_names, backend._lib.GENERAL_NAMES_free
-    )
-
-    for alt_name in san:
-        gn = _encode_general_name(backend, alt_name)
+    for name in names:
+        gn = _encode_general_name(backend, name)
         res = backend._lib.sk_GENERAL_NAME_push(general_names, gn)
         assert res != 0
 
+    return general_names
+
+
+def _encode_subject_alt_name(backend, san):
+    general_names = _encode_general_names(backend, san)
+    general_names = backend._ffi.gc(
+        general_names, backend._lib.GENERAL_NAMES_free
+    )
```

Comment:
I don't want to gc the allocation in this case because the refactored function will be used elsewhere (and needs to not be gc'd).

This does bring up an inconsistency in naming (our `_encode_<extension>` functions return a tuple of a char * and length and this new one returns a `sk_GENERAL_NAMES`). I'm happy to rename it (`_build_general_names`?) if that's helpful.